File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task1-Server/src/main/java/com/pet/hotel/data/database/entities/PetEntity.java
package com.pet.hotel.data.database.entities;

import com.pet.hotel.data.enums.PetType;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

import java.util.List;

@Data
@Entity
@Table(name = "pet")
@NoArgsConstructor
@Component
public class PetEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pet_id")
    @Setter(AccessLevel.PUBLIC)
    private int petId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "age", nullable = false)
    private int age;

    @Column(name = "weight", nullable = false)
    private float weight;

    @Enumerated(EnumType.STRING)
    @Column(name = "type_pet", nullable = false)
    private PetType typePet;

    @Column(name = "description")
    private String description;

    @Column(name = "photo_link")
    private String photoLink;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "pet")
    private List<RentEntity> rents;


    public PetEntity(String name, int age, float weight, PetType typePet, String description, String photoLink, UserEntity user, List<RentEntity> rents) {
        this.name = name;
        this.age = age;
        this.weight = weight;
        this.typePet = typePet;
        this.description = description;
        this.photoLink = photoLink;
        this.user = user;
        this.rents = rents;
    }
}

File: apzkr-pzpi-21-11-kokhanevych-pavlo/Task4-MobileClient/src/main/java/com/animal/hotel/data/user/entities/PetEntity.kt
package com.animal.hotel.data.user.entities

import com.animal.hotel.utils.enums.PetType
import com.animal.hotel.domain.models.user.Pet
import com.google.gson.annotations.SerializedName


data class PetEntity(
    val petId: Int,
    val name: String,
    val age: Int,
    val weight: Float,
    val typePet: PetType,
    val description: String?,
    val photoLink: String?,
    val user: UserEntity
) {
    fun toPet(): Pet {
        return Pet(petId, name, age, weight, typePet, description, photoLink, user.toUser())
    }

    companion object {
        fun toPetEntity(pet: Pet) = PetEntity (
            pet.petId,
            pet.name,
            pet.age,
            pet.weight,
            pet.typePet,
            pet.description,
            pet.photoLink,
            UserEntity.toUserEntity(pet.user)
        )
    }
}